source/blog/_posts/2016-10-12-promises.md

```diff
@@ -0,0 +1,353 @@
+	notify = notify || function() {};
+	
+	timer = window.setInterval(function() {
+		notify.apply(null);
+	}, 1000);
+	
+	window.setTimeout(function() {
+		window.clearInterval(timer);
+		cb.apply(null);
+	}, time);
+};
+```
+
+The above code can be used like that:
+
+```
+$('button').on('click', function() {
+	var $waitingTimer = $('.waiting-timer');
+	
+    waitingTimer(5000, function() {
+    	$waitingTimer.html($waitingTimer.html() + ".");
+  	}, function() {
+    	$waitingTimer.html("Done!");
+  	});
+});
+```
+
+<iframe src="https://jsfiddle.net/klarstil/m1p89frr/1/embedded/result/" frameborder="0" width="100%" height="250px"></iframe>
+
+Now let us promisify the above code:
+
+```
+var waitingTimer = function(time) {
+	
+	// Get the global deferred object
+	var deferred = $.Deferred(),
+		timer;
+
+	// Make sure our values are defined.
+	time = time || 10000;
+	
+	timer = window.setInterval(function() {
+		// Fire the notify method on the deferred object
+		deferred.notify();
+	}, 1000);
+	
+	window.setTimeout(function() {
+		window.clearInterval(timer);
+		
+		// Resolve the promise
+		deferred.resolve();
+	}, time);
+	
+	// We're returning the promise at that point
+	return deferred.promise();
+};
+```
+
+The promisied code can be used like that:
+
+```
+$('button').on('click', function() {
+	var $waitingTimer = $('.waiting-timer');
+	
+    waitingTimer(5000).progress(function() {
+    	$waitingTimer.html($waitingTimer.html() + ".");
+  	}).done(function() {
+    	$waitingTimer.html("Done!");
+    });
+});
+```
+
+<iframe src="https://jsfiddle.net/klarstil/jwbxt71j/embedded/result/" frameborder="0" width="100%" height="250px"></iframe>
+
+As a side note: Quite a few jQuery methods are using promises or the Deferred object already. One of the most common one is `$.ajax()` which allows to use a promise instead of callback methods:
+
+```
+$.ajax({
+	url: "/ServerResource.txt"
+}).done(result, function(result) {
+	console.log("Success!", result);
+}).fail(function(err) {
+	console.error('Failed!', err);
+});
+```
+
+jQuery uses the [CommonJS Promises/A interface](http://wiki.commonjs.org/wiki/Promises/A) for the jQuery XMLHttpRequest.
+
+There's just one downside you have to consider when working with promises or the Deferred object. jQuery's promises are linked to a Deferred object stored on the `.data()` for an element. Since the `.remove()` method removes the element's data as well as the element itself, it will prevent any of the element's unresolved Promises from resolving.
+
+## Conclusion
+Promises are looking like a brand new technology but apparently they are not. jQuery introduced them with version 1.5 which was released back in 2011. They provide a great and easy to use way to overcome the problem of having nesting callbacks in your application. As we saw in the blog post it's very easy to transform your callback code into a promised based approach which provides a higher flexibility and an easier to maintain code base.
```

Comment:
Maybe use "easier maintainable code base" instead of "easier to maintain code base".
Not sure about this though